packages/synapse-react-client/src/components/JSONArrayEditor/JSONArrayEditor.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useCallback, useState } from 'react'
+import type RJSFForm from '@rjsf/core'
+import Form from '@rjsf/mui'
+import validator from '@rjsf/validator-ajv8'
+import { JSONSchema7 } from 'json-schema'
+import ArrayFieldDescriptionTemplate from '../SchemaDrivenAnnotationEditor/template/ArrayFieldDescriptionTemplate'
+import ArrayFieldItemTemplate from '../SchemaDrivenAnnotationEditor/template/ArrayFieldItemTemplate'
+import ArrayFieldTemplate from '../SchemaDrivenAnnotationEditor/template/ArrayFieldTemplate'
+import ArrayFieldTitleTemplate from '../SchemaDrivenAnnotationEditor/template/ArrayFieldTitleTemplate'
+import ButtonTemplate from '../SchemaDrivenAnnotationEditor/template/ButtonTemplate'
+import { Box, Button, Collapse, TextField } from '@mui/material'
+import { parse } from 'papaparse'
+import { RJSFSchema } from '@rjsf/utils'
+type JSONArrayEditorProps = {
+  value: string[]
+  onChange: (newValue: string[]) => void
+  onSubmit: (formData: string[]) => void
+}
+
+const arraySchema: JSONSchema7 = {
+  $schema: 'http://json-schema.org/draft-07/schema#',
+  type: 'array',
+  items: {
+    type: 'string',
+  },
+}
+
+const JSONArrayEditor = React.forwardRef(function JSONArrayEditor(
+  props: JSONArrayEditorProps,
+  ref: React.Ref<RJSFForm<any, RJSFSchema, any>>,
+) {
+  const { value, onChange, onSubmit } = props
+  const [showPasteNewValuesForm, setShowPasteNewValuesForm] = useState(false)
+  const [pastedValues, setPastedValues] = useState('')
+
+  const addPastedValuesToArray = useCallback(() => {
+    if (pastedValues) {
+      parse<string[]>(pastedValues, {
+        complete: result => {
+          console.log(result)
+          if (result.errors.length > 0) {
+            console.error('Error parsing pasted values', result.errors)
```

Comment:
Rather show the error in an alert, and do not clear the user pasted values (and keep the form open) so user can fix the problem and retry

Reply:
<img width="911" alt="image" src="https://github.com/Sage-Bionetworks/synapse-web-monorepo/assets/17580037/2966f5b7-9e29-4e64-afa0-72320cbbac2f">